includes/polling/os/fortios.inc.php

```diff
@@ -1,3 +1,25 @@
 <?php
 
-$serial = snmp_get($device, 'fnSysSerial', '-OQv', 'FORTINET-FORTIANALYZER-MIB');
+use LibreNMS\RRD\RrdDefinition;
+
+$serial = snmp_get($device, 'FORTINET-CORE-MIB::fnSysSerial.0', '-Ovq');
+$version = snmp_get($device, 'FORTINET-FORTIMANAGER-FORTIANALYZER-MIB::fmSysVersion.0', '-Ovq');
+$hardware = rewrite_fortinet_hardware($device['sysObjectID']);
+if ($hardware == $device['sysObjectID']) {
+    unset($hardware);
+}
+
+//Log rate only for FortiAnalyzer features enabled FortiManagers
+$entModes = snmpwalk_cache_oid($device, 'fmDeviceEntMode', null, 'FORTINET-FORTIMANAGER-FORTIANALYZER-MIB');
```

Comment:
How big is this walk? the idea in polling is to minimise the number of snmp calls. Can you not just test using `$log_rate = snmp_get($device, '.1.3.6.1.4.1.12356.103.2.1.9.0', '-Ovq');` seeing as that's all you care about?

Reply:
FortiManager and FortiAnalyzer are separate products, but you can enable most of the Analyzer features inside Manager. So the idea is to poll the log rate only if those Analyzer features have been enabled. There is no specific oid for that. 

That code checks if there are any Administrative Domains in fmg-faz mode. But you are right, in very large deployments that walk can get big. I think it should be safe to just get the first ADOM's mode and check if there is 'faz' in it.

Reply:
Or just do the log_rate snmp query. Running one snmp query to determine if we should run another is the same amount of time as just running the second one only and in fact will be slower if the first query tells us we should then run the second.

In short, you should only do what you're doing if we need the data from both queries which it doesn't look like we do here.

Reply:
Log rate is only relevant if those Analyzer features are enabled. The log rate oid is present are those features enabled or not. So if we don't check that in any way, FortiManagers without analyzer features will get the log rate graph also. And its going to be constant 0. 

I understand that we want to keep snmp queries to the minimum. If you want it to be like you suggested, I will do it.

Reply:
Can the value of the log rate oid be 0 if it's enabled still? If not then you can check if it's > 0. If it can then you will need to find the best way to do a simple snmp get for the initial test, walking a tree which could be large shouldn't be done in polling.